core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt

```diff
@@ -7,44 +7,36 @@ import kotlin.reflect.KVisibility.INTERNAL
 import kotlin.reflect.KVisibility.PRIVATE
 import kotlin.reflect.KVisibility.PROTECTED
 import kotlin.reflect.KVisibility.PUBLIC
-import kotlin.reflect.full.memberProperties
 import kotlin.reflect.jvm.isAccessible
 import kotlin.reflect.jvm.javaType
 
 private const val indentBlock = "  "
 
-fun Node.relevantMemberProperties(withPosition: Boolean = false, withNodeType: Boolean = false) =
-    this.javaClass.kotlin.memberProperties
-        .filter {
-            !it.name.startsWith("component") &&
-                (it.name != "position" || withPosition) &&
-                (it.name != "nodeType" || withNodeType) &&
-                it.name != "properties" &&
-                it.name != "positionOverride" &&
-                it.name != "origin" &&
-                it.name != "destination" &&
-                it.name != "sourceText" &&
-                it.name != "source" &&
-                it.name != "parent"
-        }
+fun Node.relevantMemberProperties(withPosition: Boolean = false, withNodeType: Boolean = false):
+    List<KProperty1<Node, *>> {
+    val list = this.javaClass.kotlin.nodeProperties.toMutableList()
```

Comment:
I think we can simplify this to `this::class`

Reply:
Right, changing that